src/Hl7.Fhir.Shims.Base/Model/ValueSetExpansionExtensions.cs

```diff
@@ -67,27 +67,23 @@ internal static List<CSDC> RemoveCode(this ICollection<CSDC> concepts, string co
 
         private static ICollection<CSDC> removeCodeByPredicate(this ICollection<CSDC> concepts, Predicate<CSDC> predicate)
         {
-            foreach (var concept in concepts)
+            var itemsToBeRemoved = concepts.Where(c => predicate(c)).ToList();
```

Comment:
We iterate 3 times now. We can get away with only doing this once, by just putting all the calls in a foreach loop, or better yet: a select (if we are going pure)

Reply:
I was hoping to do this in a single forloop (or linq statement), but I couldn't figure it out without duplicating collections, and making it a mess. 
You have a suggestion?

Reply:
In this case duplicating subsets of collections is a fact of life I think. The most optimized version is a big chain call, since the compiler can chain the methods together into a single loop. I will suggest a commit (once I find out how to)

Reply:
Alright, it seems like i'm not able to suggest it the way I wanted to. I would like to suggest something along the lines of:
```cs
internal static List<CSDC> RemoveCode(this ICollection<CSDC> concepts, string code)
{
    return concepts.getNonMatchingCodes(c => c.Code == code);
}

private static List<CSDC> getNonMatchingCodes(this ICollection<CSDC> concepts, Predicate<CSDC> predicate)
{
    var filtered = concepts.Where(c => !predicate(c)).ToList();
    
    foreach (var concept in filtered.Where(concept => concept.Concept?.Count is > 0))
    {
        concept.Concept = concept.Concept.getNonMatchingCodes(predicate).ToList();
    }
    
    return filtered;
}